src/state_machine/coordinator/fire.rs

```diff
@@ -1117,14 +1164,14 @@ impl<Aggregator: AggregatorTrait> Coordinator<Aggregator> {
     fn compute_dkg_public_size(&self) -> u32 {
         self.dkg_public_shares
             .keys()
-            .map(|signer_id| self.config.signer_key_ids[signer_id].len() as u32)
+            .map(|signer_id| self.config.public_keys.signer_key_ids[signer_id].len() as u32)
             .sum()
     }
 
     fn compute_dkg_private_size(&self) -> u32 {
         self.dkg_private_shares
             .keys()
-            .map(|signer_id| self.config.signer_key_ids[signer_id].len() as u32)
+            .map(|signer_id| self.config.public_keys.signer_key_ids[signer_id].len() as u32)
```

Comment:
We should forgo the direct array access.

Reply:
Fixed in a3c5c3276cb9336959caf47f9842389d9f9beb39